src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/WriteFinalDecisionIt.java

```diff
@@ -119,13 +120,69 @@ public void callToMidEventPreviewFinalDecisionCallback_willPreviewTheDocument()
+
+        String documentUrl = "document.url";
+        when(generateFile.assemble(any())).thenReturn(documentUrl);
+
+        when(userDetails.getFullName()).thenReturn("Judge Full Name");
+
+        when(idamClient.getUserDetails("Bearer userToken")).thenReturn(userDetails);
+
+        MockHttpServletResponse response = getResponse(getRequestWithAuthHeader(json, "/ccdMidEventPreviewFinalDecision"));
+        assertHttpStatus(response, HttpStatus.OK);
+        PreSubmitCallbackResponse<SscsCaseData> result = deserialize(response.getContentAsString());
+
+        assertEquals(Collections.EMPTY_SET, result.getErrors());
+
+        assertEquals(documentUrl, result.getData().getWriteFinalDecisionPreviewDocument().getDocumentUrl());
+
+        ArgumentCaptor<GenerateFileParams> capture = ArgumentCaptor.forClass(GenerateFileParams.class);
+        verify(generateFile).assemble(capture.capture());
+        final DirectionOrDecisionIssuedTemplateBody parentPayload = (DirectionOrDecisionIssuedTemplateBody) capture.getValue().getFormPayload();
+        final WriteFinalDecisionTemplateBody payload = parentPayload.getWriteFinalDecisionTemplateBody();
+
+        assertEquals("An Test", parentPayload.getAppellantFullName());
+        assertEquals("12345656789", parentPayload.getCaseId());
+        assertEquals("JT 12 34 56 D", parentPayload.getNino());
+        assertEquals(LocalDate.parse("2017-07-17"), payload.getHeldOn());
+        assertEquals("Chester Magistrate's Court", payload.getHeldAt());
+        assertEquals("Judge Full Name, Panel Member 1 and Panel Member 2", payload.getHeldBefore());
+        assertEquals(true, payload.isAllowed());
+        assertEquals(true, payload.isSetAside());
+        assertEquals("2018-09-01", payload.getDateOfDecision());
+        assertEquals("An Test",payload.getAppellantName());
+        assertNull(payload.getStartDate());
+        assertNull(payload.getEndDate());
+        assertEquals(true, payload.isIndefinite());
+        assertEquals(false, payload.isDailyLivingIsEntited());
+        assertEquals(false, payload.isDailyLivingIsSeverelyLimited());
+        assertNull(payload.getDailyLivingAwardRate());
+        assertNull(payload.getDailyLivingDescriptors());
+        Assert.assertNull(payload.getDailyLivingNumberOfPoints());
+        assertEquals(false, payload.isMobilityIsEntited());
+        assertEquals(false, payload.isMobilityIsSeverelyLimited());
+        assertNull(payload.getMobilityAwardRate());
+        assertNull(payload.getMobilityDescriptors());
+        assertNotNull(payload.getReasonsForDecision());
+        assertEquals(1, payload.getReasonsForDecision().size());
+        Assert.assertEquals("My reasons for decision", payload.getReasonsForDecision().get(0));
+        assertNotNull(payload.getDetailsOfDecision());
+        Assert.assertEquals("The details of the decision", payload.getDetailsOfDecision());
+        assertEquals("Something else.", payload.getAnythingElse());
+    }
+
```

Comment:
Can you add a test to this class for the callToAboutToSubmitHandler_willWriteDraftFinalDecisionToCaseForNonDescriptorRoute route?

Reply:
Will do - thanks Jack